### PR TITLE
Feature: Spell check

### PR DESCRIPTION
**Pull request by DanielSiepmann**
Allow elasticsearch to suggest spelling correction.

Also introduce new concept of dataProcessing as done in TYPO3.

### CODE REVIEW

File: Classes/DataProcessing/ProcessorInterface.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * All DataProcessing Processor should implement this interface, otherwise they
```
Comment (justusmoroni) [inline Classes/DataProcessing/ProcessorInterface.php]: `Processor` plural form is needed here, right?

File: Classes/Domain/Search/QueryFactory.php [hunk 2]
```diff
@@ -214,4 +217,44 @@ protected function addFacets(SearchRequestInterface $searchRequest)
...
+
+        $suggest = new \Elastica\Suggest();
+        foreach ($suggests as $suggestToAdd) {
+            $suggest->addSuggestion($suggestToAdd);
+        }
+        $query->setSuggest($suggest);
+    }
+
+    /**
+     * @param SearchRequestInterface $searchRequest
```
Comment (justusmoroni) [inline Classes/Domain/Search/QueryFactory.php:246]: What is returned here?

File: Classes/Domain/Search/QueryFactory.php [hunk 3]
```diff
@@ -214,4 +217,44 @@ protected function addFacets(SearchRequestInterface $searchRequest)
             ]);
         }
     }
+
+    /**
+     * @param \Elastica\Query $query
+     * @param SearchRequestInterface $searchRequest
+     */
```
Comment (justusmoroni) [inline Classes/Domain/Search/QueryFactory.php:227]: What is returned here?

File: Documentation/source/configuration.rst [hunk 4]
```diff
@@ -36,296 +36,14 @@ Here is the example default configuration that's provided through static include
 Options
 -------
 
-The following section contains the different options, e.g. for :ref:`connections` and
-:ref:`indexer`: ``plugin.tx_searchcore.settings.connection`` or
-``plugin.tx_searchcore.settings.indexing``.
+The following sections containsthe different options grouped by their applied area, e.g. for
```
Comment (justusmoroni) [inline Documentation/source/configuration.rst]: Missing whitespace between `contains` and `the`.

**Review by justusmoroni — CHANGES_REQUESTED** [review]

File: Classes/Domain/Search/QueryFactory.php (hunk 3 above)
Comment (DanielSiepmann) [reply Classes/Domain/Search/QueryFactory.php:227]: Nothing, like most other methods in this class.

File: Classes/Domain/Search/QueryFactory.php (hunk 2 above)
Comment (DanielSiepmann) [reply Classes/Domain/Search/QueryFactory.php:246]: I've added the return type.

**Review by justusmoroni — APPROVED** [review]

**Comment by DanielSiepmann** [comment]
This feature didn't make it in time.

We do not try to finish it yet and therefore close it.

**Pull request closed without merging**